tools/lkl/lib/hijack/hijack.c

```diff
@@ -291,7 +304,25 @@ int select(int nfds, fd_set *r, fd_set *w, fd_set *e, struct timeval *t)
 	return lkl_call(__lkl__NR_select, 5, nfds, r, w, e, t);
 }
 
-HOOK_CALL_USE_HOST_BEFORE_START(epoll_create);
+HOST_CALL(epoll_create);
+int epoll_create(int flags)
```

Comment:
Hi Hajime, Dear Reviewers,

The previous version, i.e. with HOOK_CALL_USE_HOST_BEFORE_START, works when using lkl pure. However, it is necessary to use HOST_CALL for the dpdk version. With dpdk and HOOK_CALL_USE_HOST_BEFORE_START I get the message

/home/octavian/Programs/Nodes/node_system_calls/node/install/usr/local/bin/node: symbol lookup error: /home/octavian/Octavian/DriveNetwork/2017/07July/28/lkl-linux/tools/lkl/bin/../liblkl-hijack.so: undefined symbol: host_epoll_create

This means most probably that epoll_create is necessary on the host version.

Please let me know if this answers your question.

Best regards,
  Octavian (Soldea)